From 405a51b4ea2629409408315363773e7661a82e54 Mon Sep 17 00:00:00 2001
From: Portisch <hugo.portisch@yahoo.de>
Date: Wed, 24 Sep 2025 14:34:57 +0200
Subject: [PATCH] avformat/matroska: Support VVC for demuxing

---
 libavformat/matroska.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libavformat/matroska.c b/libavformat/matroska.c
index d0ecfbe..bbad9a7 100644
--- a/libavformat/matroska.c
+++ b/libavformat/matroska.c
@@ -90,6 +90,7 @@ const CodecTags ff_mkv_codec_tags[]={
     {"V_MPEG4/ISO/SP"   , AV_CODEC_ID_MPEG4},
     {"V_MPEG4/ISO/AVC"  , AV_CODEC_ID_H264},
     {"V_MPEGH/ISO/HEVC" , AV_CODEC_ID_HEVC},
+    {"V_MPEGI/ISO/VVC"  , AV_CODEC_ID_VVC},
     {"V_MPEG4/MS/V3"    , AV_CODEC_ID_MSMPEG4V3},
     {"V_PRORES"         , AV_CODEC_ID_PRORES},
     {"V_REAL/RV10"      , AV_CODEC_ID_RV10},
-- 
2.43.0

